From 44375bece46e77970b9c3c9a457c02dee31bfd57 Mon Sep 17 00:00:00 2001
Message-Id: <patch-v2-3.5-44375bece46-20211005T165801Z-avarab@gmail.com>
In-Reply-To: <cover-v2-0.5-00000000000-20211005T165801Z-avarab@gmail.com>
References: <cover-v2-0.5-00000000000-20211005T165801Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Fri, 1 Oct 2021 18:32:02 +0200
Subject: [PATCH v2 3/5] COPYING: remove mention of GPLv3, clarify sub-project
 COPYING
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The COPYING file added in [1] is a product of its era, at the time
discussions around the GPLv3 were quite active. Since then we have had
almost no "GPL v2 or later" notices anywhere.

Since then[2][3] linux.git has replaced its version of almost the
identical wording with something that's a bit more formal, and which
isn't mostly addressing a hypothetical future GPL v3 migration. Let's
do the same. Now we:

A. Let it suffice to say that things are licensed under GPL v2 unless
   explicitly otherwise stated. The new wording is an amalgamation of
   the start of Linux's "COPYING" file[4] and the end of our pre-image
   first paragraph added in[1].

B. Add some brief wording about how the licenses of subdirectories
   such as "sha1dc/" and "compat/regex/" should be discovered. This is
   also intended to cover future in-tree additions such as the
   "reftable" directory[5][6].

C. Include the same "subject to this COPYING file" wording as
   linux.git has used since[7].

1. 075b845a857 (Add a COPYING notice, making it explicit that the
   license is GPLv2., 2005-04-11)
2. torvalds/linux@bf02d491237e (COPYING: use the new text with points
   to the license files, 2018-03-23)
3. torvalds/linux@74835c7db032 (COPYING: state that all contributions
   really are covered by this file, 2020-02-06)
4. torvalds/linux@88de995ccf48 (COPYING: create a new file with points
   to the Kernel license files, 2018-03-23)
5. https://lore.kernel.org/git/2597ddfeae1d9da8695151f62379c85621ad0fcb.1632841817.git.gitgitgadget@gmail.com/
6. https://lore.kernel.org/git/xmqqtui4k1j5.fsf@gitster.g/
7. torvalds/linux@74835c7db032 (COPYING: state that all contributions
   really are covered by this file, 2020-02-06)

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 COPYING | 35 +++++++++++++++++------------------
 1 file changed, 17 insertions(+), 18 deletions(-)

diff --git a/COPYING b/COPYING
index 536e55524db..19de8f2ea1b 100644
--- a/COPYING
+++ b/COPYING
@@ -1,21 +1,20 @@
-
- Note that the only valid version of the GPL as far as this project
- is concerned is _this_ particular version of the license (ie v2, not
- v2.2 or v3.x or whatever), unless explicitly otherwise stated.
-
- HOWEVER, in order to allow a migration to GPLv3 if that seems like
- a good idea, I also ask that people involved with the project make
- their preferences known. In particular, if you trust me to make that
- decision, you might note so in your copyright message, ie something
- like
-
-	This file is licensed under the GPL v2, or a later version
-	at the discretion of Linus.
-
-  might avoid issues. But we can also just decide to synchronize and
-  contact all copyright holders on record if/when the occasion arises.
-
-			Linus Torvalds
+Git is provided under the terms of the GNU General Public License
+version 2 only (GPL-2.0), unless explicitly otherwise stated.
+
+The license described in this COPYING file applies to Git's source as
+a whole, though individual source files can have a different license
+which is required to be compatible with the GPL-2.0. All contributions
+to Git are subject to this COPYING file.
+
+Different but compatible licenses may apply to individual files or
+directories. Defer to files matching "*COPYING*" or "*LICENSE*" in
+subdirectories for the license of files therein, or to copyright
+notices at the top of individual files.
+
+Examples include "sha1dc/LICENSE.txt" covering the imported "sha1dc/"
+directory (and optional "sha1collisiondetection/" submodule), the
+"xdiff/" directory licensed under the LGPL v2.1 (see the "LGPL-2.1"
+file), and imported external code such as "compat/regex/".
 
 ----------------------------------------
 
-- 
2.33.0.1441.gbbcdb4c3c66

